addons/website_event_exhibitor/static/src/js/event_sponsor_search.js

```diff
@@ -10,13 +10,30 @@ publicWidget.registry.websiteEventSearchSponsor = publicWidget.Widget.extend({
         'click .o_search_tag .btn': '_onTagRemove',
         'click .o_dropdown_reset_tags': '_onTagReset',
         'change .o_wevent_event_tags_form input': '_onTagAdd',
+        'input .o_wevent_event_search_box': '_onClearInput',
+        'keyup .o_wevent_event_search_box': '_onKeyUp',
     },
 
     start: function () {
         this.form = this.$el.find('.o_wevent_event_tags_form');
         return this._super.apply(this, arguments);
     },
 
+    _onClearInput: function() {
+        const input = this.$el.find('.o_wevent_event_search_box input');
+        const url = window.location.href.split('?')[0]
+        if(!input.val()) {
+            this.form.attr('action', url);
+            this.form.submit();
+        }
```

Comment:
Hello @Odoonan 

> if (!input.val()) {
>     this._onSearch();
> }
> 

By doing this, the entire URL with the search parameters will be submitted as the form. For example, after one search, our URL will appear as follows: `event/openwood-collection-online-reveal-8/exhibitors?search=Tree`. If we clear the search input, the _onSearch() method will be called, which takes the value from the input box and creates search parameters. In our case, the value is empty, so it will submit the form with the current URL. However, to go to the home page, we need to submit the form with the `event/openwood-collection-online-reveal-8/exhibitors` URL. I'm not entirely certain if this should be considered a fix, so what should be my next step?

Thanks for the validation! 😊

Reply:
Well, if I am not mistaken the effect will be the same as pressing the search button without input, we will get something like `http://localhost:8069/event/openwood-collection-online-reveal-8/exhibitors?search=` which basically brings us back at the event home page, no? Would it be an issue? :)

Reply:
Ok I see your point. We can keep it your way (less operations in  the end). But should we not use `window.location.pathname` instead of the split, as in _onSearch ? :) What do you think?

Reply:
Hello @thsh-odoo ,

Regarding the fix, I'm greening :)

I'm still not certain the whole diff qualifies as a fix, we will see at review.

Thanks for your updates & Cheers!

Reply:
Okay @Odoonan 
I'm not entirely sure if this is a fix or an improvement either :sweat_smile:
Thanks for the validation :blush: